File: adventofcode/2024/03_1.cpp
#include <bits/stdc++.h>
#define int long long

using namespace std;

signed main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    #ifndef ONLINE_JUDGE 
        freopen("../input.txt", "r", stdin); 
        freopen("../output.txt", "w", stdout); 
    #endif
    const int LINES=6;
    std::regex pattern("mul\\((\\d+),(\\d+)\\)");

    int ans = 0;
    for(int i=0; i<LINES; i++)
    {
        string s; getline(cin, s);
        auto begin = std::sregex_iterator(s.begin(), s.end(), pattern);
        auto end = std::sregex_iterator();

        while (begin != end) {
            std::smatch match = *begin;

            if (match.size() >= 3) {
                int a = stoi(match[1]);
                int b = stoi(match[2]);
                ans += a*b;
            }

            ++begin; // Move to the next match
        }
    }
    cout << ans;
    return 0;
}

File: adventofcode/2024/python/03_1.py
import sys

import re

def main():
    sys.stdin = open("../input.txt", "r")
    sys.stdout = open("../output.txt", "w")
    
    pattern = re.compile(r"mul\((\d+),(\d+)\)")
    LINES = 6
    ans = 0
    
    for _ in range(LINES):
        try:
            s = input().strip()
        except EOFError:
            break
        if not s:
            continue
        matches = pattern.finditer(s)
        
        for match in matches:
            if len(match.group()) >= 2:
                a = int(match.group(1))
                b = int(match.group(2))
                ans += a*b
    
    print(ans)

if __name__ == "__main__":
    main()